api/src/main/kotlin/edu/wgu/osmt/collection/Collection.kt

```diff
@@ -76,6 +76,39 @@ data class CollectionUpdateObject(
         applySkills()
     }
 
+    fun applyStatusChange(dao: CollectionDao) {
+        when (publishStatus) {
+            PublishStatus.Archived ->  {
+                dao.archiveDate = LocalDateTime.now(ZoneOffset.UTC)
+                dao.status = PublishStatus.Archived
+            }
+            PublishStatus.Published -> {
+                dao.publishDate = LocalDateTime.now(ZoneOffset.UTC)
+                dao.status = PublishStatus.Published
+                dao.archiveDate = null
+            }
+            PublishStatus.Unarchived -> {
+                if (dao.publishDate != null) {
+                    dao.status = PublishStatus.Published
+
+                } else {
+                    dao.status = PublishStatus.Draft
+                }
+                dao.archiveDate = null
+            }
+            PublishStatus.Draft -> {
+                dao.status = PublishStatus.Draft
+                dao.archiveDate = null
+                dao.publishDate = null
+
+            }
+            PublishStatus.Workspace -> {
+                dao.status = PublishStatus.Workspace
+            }
+            else -> {}
```

Comment:
Can this case ever happen? CollectionDao.status is not nullable, and the dao argument is also not nullable.